modules/system/defaults/controlcenter.nix

```diff
@@ -1,100 +1,397 @@
+      } 12;
+      description = ''
+        Apple menu > System Preferences > Control Center > Accessibility Shortcuts
 
-            Show a battery percentage in menu bar. Default is null.
-        '';
+        Options:
+          both           - Show in Menu Bar and Control Center (3)
+          menuBar        - Show in Menu Bar only (6)
+          controlCenter  - Show in Control Center only (1 or 9)
+          hide           - Don't Show (2, 4, 8, or 12)
+      '';
     };
 
-    system.defaults.controlcenter.Sound = lib.mkOption {
-        type = lib.types.nullOr lib.types.bool;
-        apply = v: if v == null then null else if v == true then 18 else 24;
-        default = null; 
-        description = ''
-            Apple menu > System Preferences > Control Center > Sound
+    system.defaults.controlcenter.AirDrop = lib.mkOption {
+      type = lib.types.nullOr lib.types.bool;
+      default = null;
+      apply = mkBoolApply {
+        true = 18;
+        false = 24;
+      };
+      description = ''
+        Apple menu > System Preferences > Control Center > AirDrop
 
-            Show a sound control in menu bar . Default is null.
+        Show a AirDrop control in menu bar. Default is null.
 
-            18 = Display icon in menu bar
-            24 = Hide icon in menu bar
-        '';
+        18 = Show in Menu Bar
+        24 = Don't Show in Menu Bar
+      '';
     };
 
-    system.defaults.controlcenter.Bluetooth = lib.mkOption {
-        type = lib.types.nullOr lib.types.bool;
-        apply = v: if v == null then null else if v == true then 18 else 24;
-        default = null; 
-        description = ''
-            Apple menu > System Preferences > Control Center > Bluetooth
+    system.defaults.controlcenter.Battery = lib.mkOption {
+      type = lib.types.nullOr (
+        lib.types.enum [
+          "both"
+          "menuBar"
+          "controlCenter"
+          "hide"
+        ]
+      );
+      default = null;
+      apply = mkEnumApply {
+        both = 3;
+        menuBar = 4;
+        controlCenter = 9;
+        hide = 12;
+      } 4;
+      description = ''
+        Apple menu > System Preferences > Control Center > Battery
 
-            Show a bluetooth control in menu bar. Default is null.
+        Options:
+          both           - Show in Menu Bar and Control Center (3)
+          menuBar        - Show in Menu Bar only (6)
+          controlCenter  - Show in Control Center only (1 or 9)
+          hide           - Don't Show (2, 4, 8, or 12)
+      '';
+    };
 
-            18 = Display icon in menu bar
-            24 = Hide icon in menu bar
-        '';
+    system.defaults.controlcenter.BatteryShowEnergyMode = lib.mkOption {
+      type = lib.types.nullOr lib.types.bool;
+      default = null;
+      description = ''
+        Apple menu > System Preferences > Control Center > Battery
+
+        Show a battery energy mode. Default is null.
+
+        false - "When Active"
+        true  - "Always"
```

Comment:
Maybe don't use a boolean for this? an enum `whenActive` / `always` would be clearer.
OR rename the option like `...AlwaysShowBatteryEnergyMode` so the boolean makes sense 🤔